packages/std/src/os/kernel.rs

```diff
@@ -215,3 +216,338 @@ pub struct RefundData {
     pub funds: Coin,
     pub channel: String,
 }
+
+use crate::common::reply::ReplyId;
+use crate::error::ContractError;
+use cosmwasm_std::{attr, BankMsg, CosmosMsg, DepsMut, SubMsg, WasmMsg};
+use cw20::Cw20ExecuteMsg;
+
+/// Creates a CW20 send message with an attached message payload and its associated attributes.
+///
+/// This differs from `create_cw20_transfer_msg` in that it sends tokens to a contract with an
+/// attached message that the receiving contract can process.
+///
+/// # Arguments
+///
+/// * `contract` - The contract address that will receive the tokens and process the message
+/// * `token_addr` - The CW20 token contract address
+/// * `amount` - The amount of tokens to send
+/// * `msg` - The message to attach to the token transfer
+/// * `config` - Optional configuration for the submessage (gas limit, reply behavior)
+///
+/// # Returns
+///
+/// Returns a tuple containing:
+/// * `SubMsg` - A submessage configured to send the CW20 tokens with the specified message
+/// * `Vec<Attribute>` - A vector of attributes documenting the token send
+pub fn create_cw20_send_msg(
```

Comment:
Is this likely to be used by other ADOs? If so we should also provide a submessage ID parameter. If not this should probably go in the kernel contract directory.